packages/context/src/context.ts

```diff
@@ -43,7 +55,11 @@ export class Context {
    *
    * @param key Binding key
    */
-  bind(key: string): Binding {
+  bind<T = BoundValue>(key: string | BindingKey<T>): Binding<T> {
+    if (typeof key !== 'string') {
+      key = key.value;
```

Comment:
Discussion point:

I am thinking about two alternatives, I am not sure which one is better:

 - keep the current solution where we require the key object to have a `.value` property
 - use a different approach - convert the key object to a string key via `.toString()`

The difference is for people using `Context` in JavaScript codebase, it allows them to use anything as a key, as long as the value can be converted to a string.

Thoughts?

Reply:
I think the current approach is sufficient requiring a `.value` property to the key object. However, I see the UX value in us handling it internally with your alternative approach.

Reply:
One problem I see with our current approach is that two separate objects with key `value` mapping to the same string would not behave as users may expect it to.

As an aside, did you mean `JSON.stringify` as opposed to `.toString()`?

Reply:
> One problem I see with our current approach is that two separate objects with key value mapping to the same string would not behave as users may expect it to.

@shimks I am afraid I don't understand, could you please add more details? What would users expect and what would they get instead?

> As an aside, did you mean JSON.stringify as opposed to .toString()?

I meant `.toString()`. To be more precise, I would use the following conversion for its performance:

```js
key = '' + key;
```